scripts/generate-utility-data.ts

```diff
@@ -0,0 +1,253 @@
+ */
+const OVERRIDES = new Map<string | number, string>([
+  // AZ
+  [1241, 'DixiePower'],
+  [15048, 'Electrical District No 2 Pinal County'],
+  [30518, 'Electrical District No 3 Pinal County'],
+  [15049, 'Electrical District No 4 Pinal County'],
+  [18280, 'Sulphur Springs Valley Electric Cooperative'],
+  [19728, 'UniSource Energy Services'],
+  [40165, 'DixiePower'],
+  [62683, "Tohono O'odham Utility Authority"],
+  [606953, 'UniSource Energy Services'],
+
+  // CO
+  [5997, 'Estes Park Power and Communications'],
+  [7300, 'Glenwood Springs Electric'],
+  [10066, 'K.C. Electric Association'],
+  [11256, 'Loveland Water and Power'],
+  [12860, 'Morgan County REA'],
+  [15257, 'Poudre Valley REA'],
+  [16603, 'San Luis Valley REC'],
+  [16616, 'San Isabel Electric'],
+
+  // CT
+  [7716, 'Groton Utilities'],
+  [13831, 'Norwich Public Utilities'],
+  [17569, 'South Norwalk Electric and Water'],
+
+  // NY
+  [1036, 'Con Edison'],
+  [1115, 'NYSEG'],
+  [1117, 'National Grid'], // Niagara Mohawk
+  [3249, 'Central Hudson Gas & Electric'],
+  [4226, 'Con Edison'],
+  [11811, 'Massena Electric'],
+  [13511, 'NYSEG'],
+  [14154, 'Orange & Rockland'],
+  [16183, 'Rochester Gas & Electric'],
+  [16549, 'Salamanca Board of Public Utilities'],
+
+  // RI
+  [1857, 'Block Island Power Company'],
+
+  // VA
+  [1186, 'Dominion Energy'],
+  [2248, 'BVU Authority'],
+  [4794, 'Danville Utilities'],
+  [6715, 'Franklin Municipal Power and Light'],
+  [13640, 'NOVEC'],
+  [15619, 'Radford Electric Department'],
+  [19876, 'Dominion Energy'],
+  [60762, 'BVU Authority'],
+  ['Virginia Electric & Power Co', 'Dominion Energy'],
+
+  // VT
+  [1061, 'Green Mountain Power'],
+  [7601, 'Green Mountain Power'],
+  [19791, 'Vermont Electric Coop'],
+  [27316, 'Stowe Electric Department'],
+]);
+
+/**
+ * A best-effort attempt to convert the name from the spreadsheet into a
+ * utility ID and a user-facing name, as we've done manually.
+ */
+function convertName(
+  name: string,
+  state: string,
+): { id: string; name: string } {
+  let cleaned = name;
+
+  // Remove the " - (XX)" suffix (state codes)
+  cleaned = cleaned.replace(/-? *\([A-Z]{2}\)$/, '').trim();
+  // Remove square-bracketed parts, which are parent companies not used as the
+  // customer-facing brand
+  cleaned = cleaned.replace(/\[.*\]/, '').trim();
+  // Remove "Inc" or "Co" suffix
+  cleaned = cleaned.replace(/,? *(I(nc)?|Co)\.?$/, '').trim();
+  // Remove "(for __ reporting)" parenthetical
+  cleaned = cleaned.replace(/\(.*reporting.*\)/i, '').trim();
+  // Spell out "Pwr", "Assn", "Elec", and "Coop"
+  // The dataset is inconsistent about this
+  cleaned = cleaned.replaceAll(/\bPwr\b/g, 'Power');
+  cleaned = cleaned.replaceAll(/\bAssn\b/g, 'Association');
+  cleaned = cleaned.replaceAll(/\bElec\b/g, 'Electric');
+  cleaned = cleaned.replaceAll(/\bCo-?op\b/g, 'Cooperative');
+
+  // Generate the ID: replace & with "and", remove all parentheticals,
+  // spaces into hyphens
+  const id =
```

Comment:
There is an existing createAuthorityName function in `scripts/lib/authority-and-program-updater.ts`. I don't care which we use, but it's important that we have a consistent way to go from authority name to authority ID; otherwise the spreadsheet-to-json flow is going to create invalid IDs.

Reply:
Ah, OK. I'll use that but I'll add the `&` -> `and` replacement to it, since that comes up all the time in utility names and I think it makes the resulting ID clearer